EventFilter/HcalRawToDigi/plugins/PackerHelp.h

```diff
@@ -401,14 +418,37 @@ class UHTRpacker {
 
     int fiber = eid.fiberIndex();
     int fiberchan = eid.fiberChanId();
-    int capid0 = qiedf.samples() == 0 ? 0 : qiedf[0].capid();  // capacitor id for the first sample
-
-    header |= (fiberchan & QIE11HeaderSpec::MASK_FIBERCHAN) << QIE11HeaderSpec::OFFSET_FIBERCHAN;
-    header |= (fiber & QIE11HeaderSpec::MASK_FIBER) << QIE11HeaderSpec::OFFSET_FIBER;
-    header |= (capid0 & QIE11HeaderSpec::MASK_CAPID) << QIE11HeaderSpec::OFFSET_CAPID;
-    header |= (0x0 & QIE11HeaderSpec::MASK_FIBERERR) << QIE11HeaderSpec::OFFSET_FIBERERR;
-    header |= (0x0 & QIE11HeaderSpec::MASK_FLAVOR) << QIE11HeaderSpec::OFFSET_FLAVOR;  //flavor
-    header |= (0x1 & QIE11HeaderSpec::MASK_HEADER_BIT) << QIE11HeaderSpec::OFFSET_HEADER_BIT;
+    // capacitor id for the first sample
+
+    // Flavor is added
+
+    int flavor = qiedf[0].flavor();
+    int capid0;
+    if (qiedf.samples() == 0)
+      capid0 = 0;
+    if (flavor == 3)
+      capid0 = qiedf[1].capid();
+    else
+      capid0 = qiedf[0].capid();
+
+    // adding only cases for flavor 3
+
+    switch (flavor) {
+      case 0:
+        header |= (fiberchan & QIE11HeaderSpec0::MASK_FIBERCHAN) << QIE11HeaderSpec0::OFFSET_FIBERCHAN;
+        header |= (fiber & QIE11HeaderSpec0::MASK_FIBER) << QIE11HeaderSpec0::OFFSET_FIBER;
+        header |= (capid0 & QIE11HeaderSpec0::MASK_CAPID) << QIE11HeaderSpec0::OFFSET_CAPID;
+        header |= (0x0 & QIE11HeaderSpec0::MASK_FIBERERR) << QIE11HeaderSpec0::OFFSET_FIBERERR;
+        header |= (flavor & QIE11HeaderSpec0::MASK_FLAVOR) << QIE11HeaderSpec0::OFFSET_FLAVOR;  //flavor
+        header |= (0x1 & QIE11HeaderSpec0::MASK_HEADER_BIT) << QIE11HeaderSpec0::OFFSET_HEADER_BIT;
```

Comment:
Shouldn't it ```break``` here?

Reply:
Above ```flavor == 0``` is the "default": shouldn't it be the same also here?
Wouldn't be more convenient to include these into the if block above (from L429)?

Reply:
Indeed it would be more convenient to include these lines in the if block, such modifications made in commit 768d39c